aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -294,6 +298,28 @@ inline Return Dispatcher::callWithDispatchKey(const OperatorHandle& op, Dispatch
   detail::unused_arg_(args...);  // workaround for a false-positive warning about unused parameters in gcc 5
   const auto& dispatchTable = op.operatorIterator_->op.dispatch_table();
   const KernelFunction& kernel = dispatch_(dispatchTable, dispatchKey);
+  // Check if we need to run callbacks registered with RecordFunction
+  // If true and callbacks need inputs, we box the arguments and pass
+  // them into the callbacks and also into the kernel call
+
+  // Note: for perf reasons we wouldn't want to pass arguments into
+  // the function call or prematurely box them
+  at::RecordFunction guard(at::RecordScope::FUNCTION);
+  if (guard.active) {
+    if (guard.needs_inputs) {
+      std::vector<c10::IValue> stack;
```

Comment:
if this is a perf problem, we could think about using a thread_local std::vector and reuse it instead of allocating one on each call. But not sure if perf is important here, maybe not because we're inside the guard.

Reply:
"active" is generally false most of the time

Reply:
ok. Can we add C10_UNLIKELY to the if condition?